pkg/asset/installconfig/gcp/validation.go

```diff
@@ -113,6 +115,26 @@ func validateInstanceTypes(client API, ic *types.InstallConfig) field.ErrorList
 	return allErrs
 }
 
+// validateManagedZones will validate the public and private managed zones if they exist.
+func validateManagedZones(client API, ic *types.InstallConfig, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if ic.GCP.PublicDNSZone != "" && !strings.EqualFold(ic.GCP.PublicDNSZone, gcp.DNSZoneDisabled) {
+		if _, err := client.GetDNSZoneByName(context.TODO(), ic.GCP.ProjectID, ic.GCP.PublicDNSZone); err != nil {
```

Comment:
Couldn't this DNS zone be in a different project other than ic.GCP.ProjectID?

Reply:
ahh that is just a default place to look. If the full name which should be in PublicDNSZone (something in the format of /projects/{project}/managedZones/{managedZone} ). If the user does not supply the name in the long format then the ProjectID is used as default

Reply:
Oops. Yeah this is correct. I realized this as I read through the code but forgot to delete my comment.